elastic/kibana
# [Infra UI] Removed unused saved object attribute and unused source configuration code

Resolves https://github.com/elastic/kibana/issues/168240

### Changes

- Removes `fields.message` from the `infrastructure-ui-source` saved object who's value was being populated by `xpack.infra.sources.default.fields.message` from config.  See https://www.elastic.co/guide/en/kibana/master/logs-ui-settings-kb.html
- Removes `getInternalSourceConfiguration` and `defineInternalSourceConfiguration` functions introduced in https://github.com/elastic/kibana/pull/36066 as I cannot see them being used anywhere.   Stops exposing `defineInternalSourceConfiguration` as part of server plugin interface.
- Removes `getStaticDefaultSourceConfiguration` from InfraSources class as we aren't using `sources` from kibana config in source configuration anymore.  
- Removes deprecations warning of removal in 8.0 for other fields belonging to config xpack.infra.sources.* introduced in https://github.com/elastic/kibana/pull/115103
- Removes `getAllSourceConfigurations` used only in removed deprecations file https://github.com/elastic/kibana/pull/169430/commits/f427278322eced863f0ed80b23b8aa870689f94e#diff-081721894fc437938eb652beae0a0640ddeee532ec5e48af1f3093c8074f1eecL195
- Removed `getAllSavedSourceConfigurations` only used in `getAllSourceConfigurations`

### How to test 
- `infrastructure-ui-source` saved object no longer has `fields` attributes
  - in kibana.yml set `xpack.infra.sources.default.fields.message: ['testmessage', '@testmessage']`
  - go to Infra -> Settings
  - Change the name and save the form
  - view the `infrastructure-ui-source` saved object and no `fields` attribute should exist (unless it previously existed in an already existing `infrastructure-ui-source` saved object
- changes do not affect`fields.message` is used in logs_shared plugin
  -  i'm not sure how to easily test this and relying on someone with more Logs knowledge to help out, but from what I can see logs accesses the config directly (`xpack.infra.sources.default.fields.message`) and never uses the saved object field being removed.
